<launch>

  <!-- Configurations -->
  <!-- launch_openni: true if you want to include openni2_launch with custom parameters, if set to false make sure you enable depth_registration and set intrinsics parameters correctly -->
  <arg name="launch_openni" default="true" />
  <!-- asus: which asus to use. Currently there are 2 asus sensors. 'asus1' is the one at CP, 'asus2' is the one at PhaseSpace lab. This argument is needed to
       correctly load intrinsic calibrations from calibration package -->
  <arg name="asus" default="asus1" />

  <group if="$(arg launch_openni)">
    <!-- Launch openni2 with custom driver parameters -->
    <include file="$(find openni2_launch)/launch/openni2.launch">  
      <arg name="depth_registration"              value="true" />
      <arg name="color_depth_synchronization"     value="true" />
      <arg name="rgb_camera_info_url"       value="file:///$(find calibration)/config/intrinsics/rgb_$(arg asus).yaml" />
      <arg name="depth_camera_info_url"     value="file:///$(find calibration)/config/intrinsics/depth_$(arg asus).yaml" />
    </include>
  </group>
	
	<!-- the scene acquirer node  -->
	<node name="scene_filter_node" pkg="scene_filter_node" type="scene_filter_node" output="log" />

</launch>
